packages/core/getBusyTimes.ts

```diff
@@ -143,26 +144,43 @@ export async function getBusyTimes(params: {
         },
       });
 
-  const bookingSeatCountMap: { [x: string]: number } = {};
+  const bookingSeatCountMap = new Set<string>();
   const busyTimes = bookings.reduce(
     (aggregate: EventBusyDetails[], { id, startTime, endTime, eventType, title, ...rest }) => {
-      if (rest._count?.seatsReferences) {
+      if (eventType?.seatsPerTimeSlot) {
         const bookedAt = `${dayjs(startTime).utc().format()}<>${dayjs(endTime).utc().format()}`;
-        bookingSeatCountMap[bookedAt] = bookingSeatCountMap[bookedAt] || 0;
-        bookingSeatCountMap[bookedAt]++;
+        bookingSeatCountMap.add(bookedAt);
         // Seat references on the current event are non-blocking until the event is fully booked.
         if (
           // there are still seats available.
-          bookingSeatCountMap[bookedAt] < (eventType?.seatsPerTimeSlot || 1) &&
           // and this is the seated event, other event types should be blocked.
+          rest._count?.seatsReferences &&
+          rest._count.seatsReferences < eventType.seatsPerTimeSlot &&
           eventTypeId === eventType?.id
         ) {
+          if (eventType?.beforeEventBuffer) {
+            aggregate.push({
+              start: dayjs(startTime)
+                .subtract((eventType.beforeEventBuffer || 0) + (afterEventBuffer || 0), "minute")
+                .toDate(),
+              end: dayjs(startTime).toDate(),
```

Comment:
I think we also need to add buffer here. 
For example, a 15 minute event with a 1 hour before buffer: 
![Screenshot 2024-04-15 at 12 16 17 PM](https://github.com/calcom/cal.com/assets/30310907/63ca4cbe-f1ff-4c5b-a555-9472671a0b84)
12:15 can be booked but wouldn't have a 1 hour before buffer. For other event types this is applied correctly:
![Screenshot 2024-04-15 at 12 16 44 PM](https://github.com/calcom/cal.com/assets/30310907/b03cc8ab-9712-4850-9695-ee4e74f581b8)